src/components/DragBox/DragBox.js

```diff
@@ -0,0 +1,351 @@
+  static propTypes = {
+    /**
+     * Callback invoked when the mouse is clicked in the **DragBox** area
+     */
+    onMouseDown: PropTypes.func,
+    /**
+     * Callback invoked when the mouse is released in the **DragBox** area.
+     * Passed the **Set** of collision keys as its argument.
+     */
+    onMouseUp: PropTypes.func,
+    /**
+     * Callback invoked when the **DragBox** box set of collisions changes.
+     * Called with an object that has two keys, `added` and `removed`, each
+     * with an array of items that were just added or removed respectively
+     * to the collision set.
+     */
+    onCollisionChange: PropTypes.func,
+    /**
+     * Callback invoked when the **DragBox** box collides with new items
+     * that it wasn't colliding with at the previous mouse position.
+     * Called with an array of items that have just been collided with.
+     */
+    onCollisionBegin: PropTypes.func,
+    /**
+     * Callback invoked when the **DragBox** box stops colliding with items
+     * that it was colliding with at the previous mouse position.
+     * Called with an array of items that have just stopped colliding.
+     */
+    onCollisionEnd: PropTypes.func,
+    /**
+     * Render prop to render contents inside **DragBox**.
+     * Supports all callbacks that **DragBox** itself supports as well as the following props:
+     *
+     * *collisions*: The **Set** of keys that are currently being collided with
+     *
+     * *getRef*: A function that should be passed into the `ref` property of each child
+     * that can be selected by **DragBox**. It should be called with a unique key
+     * for each component. For example, `<button ref={getRef("uniquekey")} key="uniquekey" />`.
+     */
+    renderContents: PropTypes.func,
+    /**
+     * Render prop to render the draggable rectangle. It is passed an object with the left, top, width, and height of
+     * the rectangle to be drawn should return the component to render.
+     */
+    renderRect: PropTypes.func,
+    /**
+     * The selector for the element to attach the scroll listener to. Defaults to page body. Do not change after initial call.
+     */
+    scrollSelector: PropTypes.string,
+  };
+
+  state = {
+    mouseDown: false,
+    start: null,
+    end: null,
+    collisions: new Set(),
+    scrollLeft: 0,
+    scrollTop: 0,
+  };
+
+  static defaultProps = {
+    onMouseDown: noop,
+    onMouseUp: noop,
+    onCollisionChange: noop,
+    onCollisionBegin: noop,
+    onCollisionEnd: noop,
+    renderContents: noop,
+    scrollSelector: 'body',
+    renderRect: ({ left, top, width, height }) => (
+      <Rect
+        style={{
+          left,
+          top,
+          width,
+          height,
+        }}
+      />
+    ),
+  };
+
+  childElements = {};
+
+  componentDidMount = () => {
+    const el =
+      window.document.querySelector(this.props.scrollSelector) ||
+      window.document;
+    el.addEventListener('scroll', debounce(this.handleScroll), 16);
```

Comment:
Memory leak, `debounce` creates a new function wrapper. On `componentWillUnmount` we are removing `this.handleScroll` which is a different function reference. As a result the a DOM node will retain a reference to this object and the el domtree.

More succinctly:

Add a reference to `debounce(this.handleScroll)` use that reference for removing the event listener